### PR TITLE
feat: add proxy

### PR DESCRIPTION
**Pull request by yovanoc**
Hello, I add proxy support for http endpoints. Maybe we have some more to do for ws.

I add the typing for futuresCandles too, and add support for this 2 more methods:

- futuresMarkPriceCandles
- futuresIndexPriceCandles

### CODE REVIEW

File: src/http-client.js [hunk 1]
```diff
@@ -111,21 +112,26 @@ const checkParams = (name, payload, requires = []) => {
  * @param {object} headers
  * @returns {object} The api response
  */
-const publicCall = ({ endpoints }) => (path, data, method = 'GET', headers = {}) =>
-  sendResult(
+const publicCall = ({ proxy, endpoints }) => (path, data, method = 'GET', headers = {}) => {
+  const opts = {
+    method,
+    json: true,
+    headers,
```
Comment (balthazar) [inline src/http-client.js:119]: can you add the agent like so in this PR:

```
headers,
...(proxy ? { agent: new HttpsProxyAgent(proxy) } : {}),
```

File: src/http-client.js [hunk 2]
```diff
@@ -180,18 +187,23 @@ const privateCall = ({ apiKey, apiSecret, endpoints, getTime = defaultGetTime, p
 
     const newData = noExtra ? data : { ...data, timestamp, signature }
 
+    const opts = {
+      method,
+      headers: { 'X-MBX-APIKEY': apiKey },
+      json: true,
+    }
+    if (proxy) {
```
Comment (balthazar) [inline src/http-client.js:195]: same here

**Review by balthazar — CHANGES_REQUESTED** [review]

File: src/http-client.js (hunk 1 above)
Comment (yovanoc) [reply src/http-client.js:119]: Oh I thought it was cleaner but yes ok

**Comment by balthazar** [comment]
Thanks!

**Pull request merged**